test/Dialect/LLHD/Simulator/sim_reg.mlir

```diff
@@ -1,22 +1,22 @@
 // RUN: llhd-sim %s -n 10 | FileCheck %s
 
-// CHECK: 0ps 0d 0e  root/sig1  0x00000000
+// CHECK: 0ps 0d 0e  root/clock  0x00
+// CHECK-NEXT: 0ps 0d 0e  root/sig1  0x00000000
 // CHECK-NEXT: 0ps 0d 0e  root/sig2  0x00000000
-// CHECK-NEXT: 0ps 0d 0e  root/clock  0x00
 // CHECK-NEXT: 1000ps 0d 0e  root/sig1  0x00000002
 // CHECK-NEXT: 2000ps 0d 0e  root/clock  0x01
 // CHECK-NEXT: 3000ps 0d 0e  root/sig1  0x00000001
 // CHECK-NEXT: 3000ps 0d 0e  root/sig2  0xffffffff
-// CHECK-NEXT: 4000ps 0d 0e  root/sig1  0x00000003
 // CHECK-NEXT: 4000ps 0d 0e  root/clock  0x00
+// CHECK-NEXT: 4000ps 0d 0e  root/sig1  0x00000003
 // CHECK-NEXT: 5000ps 0d 0e  root/sig1  0x00000000
 // CHECK-NEXT: 5000ps 0d 0e  root/sig2  0x00000000
-// CHECK-NEXT: 6000ps 0d 0e  root/sig1  0x00000002
 // CHECK-NEXT: 6000ps 0d 0e  root/clock  0x01
+// CHECK-NEXT: 6000ps 0d 0e  root/sig1  0x00000002
 // CHECK-NEXT: 7000ps 0d 0e  root/sig1  0x00000001
 // CHECK-NEXT: 7000ps 0d 0e  root/sig2  0xffffffff
-// CHECK-NEXT: 8000ps 0d 0e  root/sig1  0x00000003
 // CHECK-NEXT: 8000ps 0d 0e  root/clock  0x00
+// CHECK-NEXT: 8000ps 0d 0e  root/sig1  0x00000003
 // CHECK-NEXT: 9000ps 0d 0e  root/sig1  0x00000000
 // CHECK-NEXT: 9000ps 0d 0e  root/sig2  0x00000000
```

Comment:
I guess this is due to the change from `SmallSet` to `SmallVec`. Is the new vector-based approach deterministic (because it's essentially in execution order)? Otherwise we might run into trouble specifying checks for trace outputs.

Reply:
This should actually only be due to the reordering of the entries (since we're finally sorting by the hierarchical paths within a step). The actual values are all still the same I think.

But otherwise, the `SmallVec` approach should be deterministic, yes, and always run the instances in the order of their names (since the vector gets sorted for uniquing).